catalog/nsili/catalog-nsili-common/src/main/java/com/connexta/alliance/nsili/common/ResultDAGConverter.java

```diff
@@ -0,0 +1,457 @@
+            }
+        }
+
+        if (metacard.getResourceSize() != null) {
+            try {
+                Double resSize = Double.valueOf(metacard.getResourceSize());
+                Double resSizeMB = convertToMegabytes(resSize);
+                addDoubleAttribute(graph, fileNode, NsiliConstants.EXTENT, resSizeMB, orb);
+            } catch (NumberFormatException nfe) {
+                LOGGER.warn("Couldn't convert the resource size to double: "
+                        + metacard.getResourceSize());
+            }
+        }
+
+        if (metacard.getContentTypeName() != null) {
+            addStringAttribute(graph,
+                    fileNode,
+                    NsiliConstants.FORMAT,
+                    metacard.getContentTypeName(),
+                    orb);
+        }
+
+        if (metacard.getContentTypeVersion() != null) {
+            addStringAttribute(graph,
+                    fileNode,
+                    NsiliConstants.FORMAT_VERSION,
+                    metacard.getContentTypeVersion(),
+                    orb);
+        }
+
+        Attribute downloadUrlAttr = metacard.getAttribute(Metacard.RESOURCE_DOWNLOAD_URL);
+        if (downloadUrlAttr != null) {
+            String downloadUrl = String.valueOf(downloadUrlAttr.getValue());
+            if (downloadUrl != null) {
+                downloadUrl = modifyUrl(downloadUrl, metacard.getTitle());
+                addStringAttribute(graph, fileNode, NsiliConstants.PRODUCT_URL, downloadUrl, orb);
+            }
+        }
+
+        if (metacard.getTitle() != null) {
+            addStringAttribute(graph, fileNode, NsiliConstants.TITLE, metacard.getTitle(), orb);
+        }
+
+        String siteName = SystemInfo.getSiteName();
+
+        boolean productLocal = true;
+        if (siteName != null && metacard.getSourceId() != null
+                && !siteName.equals(metacard.getSourceId())) {
+            productLocal = false;
+        }
+        addBooleanAttribute(graph, fileNode, NsiliConstants.IS_PRODUCT_LOCAL, productLocal, orb);
+    }
+
+    public static void addParts(DirectedAcyclicGraph<Node, Edge> graph, Node productNode,
+            Metacard metacard, ORB orb) {
+        Any any = orb.create_any();
+        Node partNode = new Node(0, NodeType.ENTITY_NODE, NsiliConstants.NSIL_PART, any);
+        graph.addVertex(partNode);
+        graph.addEdge(productNode, partNode);
+
+        //Determine if more than one part specific view is associated with data in this metacard
+        boolean partAdded = false;
+        String type = null;
+
+        addCoverageNodeWithAttributes(graph, partNode, metacard, orb);
+
+        if (metacardContainsImageryData(metacard)) {
+            type = NsiliProductType.IMAGERY.getSpecName();
+        }
+
+        if (metacardContainsGmtiData(metacard)) {
+            type = NsiliProductType.GMTI.getSpecName();
+            if (partAdded) {
```

Comment:
Once we finish the normalized taxonomy they will be properly populated. This class will also be expanded to map more fields from the normalized taxonomy to NSILI DAG